buf/registry/module/v1/label.proto

```diff
@@ -37,10 +37,7 @@ message Label {
   google.protobuf.Timestamp create_time = 2 [(buf.validate.field).required = true];
   // The last time the Label was updated on the BSR.
   google.protobuf.Timestamp update_time = 3 [(buf.validate.field).required = true];
-  // Whether or not the Label was archived.
-  //
-  // If a Label is archived, the API considers it to not exist and will not return archived
-  // Labels, except when calling LabelService.GetLabels.
```

Comment:
This has wider-ranging consequences. Have we decided to allow users to depend on labels that are archived? Can I still consume this label via Generated SDKs?

Reply:
I think those are orthogonal decisions decoupled with the API we expose. The API exposes the information and clients (including the CLI) can decide what they want to do or not do with that information.

The CLI could decide to allow, warn, or error if a module is depending on an archived label. I think any choice here could be defensible, but I don't think this is an API concern. If the CLI is currently relying on archived labels not being returned at all, and that is the behavior that we want to preserve, then we will need to update the CLI to use the correct filter on ListLabels.

AFAIK generated SDKs is an implementation detail of the BSR and doesn't use the API anyway, so no matter what this API is, generated SDKs could do whatever it wanted, right?

I think it is good to keep in mind that the original intent of archiving is purely for user presentation to hide things that are no longer relevant. Whether we should prevent archived labels from being used is a step further than what we strictly need. Given we tend to want to minimize breaking builds, and given the UI we have discussed will allow you to browse an archived label, I am inclined to not prevent dependencies on archived labels (both in CLI and Generated SDKs).

Will it cause a problem if a label cycles between archived and unarchived without a new commit getting pushed?

Reply:
The decision may be orthogonal, but you cannot change one without impacting the other. For example, how we enforce that commits can/cannot be consumed through archived labels is directly tied to API behavior:

- Before this PR, archived labels are not valid `ResourceRef`s in all but one API, so we have effectively decided that users _cannot_ consume commits through archived labels.
- With this PR, without any changes to the CLI, we are effectively taking the position that users _can_ consume commits through archived labels. 
    - In addition, if any past CLI used these APIs, we basically cannot make these changes without reversing course on our decision. This is not true today, but my point is that API changes are not in a vacuum from API consumers.

I'm not questioning the behaviour change -- that is, like you mentioned, orthogonal -- but I want to understand what the downstream impacts of this are and if any thinking has changed there. Generated SDKs is not impacted by APIs, but it's another downstream impact that I'm thinking about.

Reply:
I thought I reviewed all locations in the API the referenced archived labels, but I did miss some (e.g. on ResourceRef) so will take a look at those that I missed next.

Reply:
> Have we decided to allow users to depend on labels that are archived? Can I consume this label via Generated SDKs?

The intent moving forward is that archived labels are valid ResourceRefs, so you should continue to be able to depend on them and use them in Generated SDKs.